tests/test_interrupted_sync.py

```diff
@@ -0,0 +1,186 @@
+
+                    # Verify final bookmark matched the formatting standards for the resuming sync
+                    self.assertIsNotNone(final_stream_bookmark)
+                    self.assertIsInstance(final_stream_bookmark, str)
+
+                    account_bookmark_datetime = interrupted_sync_state["bookmarks"].get(stream, {}).get(account_id)
+
+                    if stream == interrupted_sync_state.get("currently_syncing"):
+                        # Assign the start date to the interrupted stream
+                        interrupted_stream_datetime = account_bookmark_datetime if account_bookmark_datetime \
+                            else start_date_datetime
+                        interrupted_stream_time = self.dt_to_ts(interrupted_stream_datetime)
+
+                        for record in interrupted_records[account_id]:
+                            record_time = self.dt_to_ts(record.get(list(replication_key)[0]))
+
+                            # Verify resuming sync only replicates records with the replication key
+                            # values greater or equal to the state for streams that were replicated
+                            # during the interrupted sync.
+                            self.assertGreaterEqual(record_time, interrupted_stream_time)
+
+                            # Verify the interrupted sync replicates the expected record set all
+                            # interrupted records are in full records
+                            self.assertIn(record, full_records[account_id],
+                                          msg="Incremental table record in interrupted sync not found in full sync")
+
+                        # Record count for all streams of interrupted sync match expectations
+                        records_after_interrupted_bookmark = 0
+                        for record in full_records[account_id]:
+                            record_time = self.dt_to_ts(record.get(list(replication_key)[0]))
+                            if record_time >= interrupted_stream_time:
+                                records_after_interrupted_bookmark += 1
+
+                        self.assertGreaterEqual(records_after_interrupted_bookmark, interrupted_record_count,
+                                                msg=f"Expected {records_after_interrupted_bookmark} records in each sync")
+
+                    else:
+                        # Get the date to start 2nd sync for non-interrupted streams
+                        synced_stream_bookmark = account_bookmark_datetime if account_bookmark_datetime \
+                            else start_date_datetime
+
+                        synced_stream_datetime = start_date_datetime if start_date_datetime \
+                            else self.dt_to_ts(synced_stream_bookmark)
+
+                        # Verify we replicated some records for the non-interrupted streams
+                        self.assertGreater(interrupted_record_count, 0)
```

Comment:
This can be true only if retrieved records that have `replication_key` value greater than or equal to saved bookmark value.In the code, I see, they are checking for records whose `replication key` value is only greater than bookmark value [here](https://github.com/singer-io/tap-tiktok-ads/blob/master/tap_tiktok_ads/streams.py#L173) and [here](https://github.com/singer-io/tap-tiktok-ads/blob/master/tap_tiktok_ads/streams.py#L187).
This assertion might fail if there are no records updated after bookmark value mentioned in `interrupted_sync_state`

Reply:
This has been fixed in parent branch [here](https://github.com/singer-io/tap-tiktok-ads/blob/24354e75565b0635ea8751e7772beb34a385acb4/tap_tiktok_ads/streams.py#L173)